### PR TITLE
Add proxy_version label to SD metrics

### PR DESCRIPTION
**Pull request by bochengchu**
**What this PR does / why we need it**:
Add proxy_version label to SD metrics

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by lei-tang — COMMENTED** [review]
LGTM.

File: extensions/common/proto_util.cc [hunk 1]
```diff
@@ -80,6 +80,8 @@ extractNodeFlatBufferFromStruct(const google::protobuf::Struct& metadata) {
       for (const auto& ip : ip_addresses) {
         ip_addrs.push_back(fbb.CreateString(toStdStringView(ip)));
       }
+    } else if (it.first == "PROXY_VERSION") {
+      proxy_version = fbb.CreateString(it.second.string_value());
```
Comment (kyessenov) [inline extensions/common/proto_util.cc:84]: This is duplicate with ISTIO_VERSION label I think. Are you expecting some mis-alignment?

File: extensions/common/context.cc [hunk 2]
```diff
@@ -265,7 +266,12 @@ flatbuffers::DetachedBuffer extractLocalNodeFlatBuffer() {
       ip_addrs.push_back(fbb.CreateString(toStdStringView(ip)));
     }
   }
-
+  if (getValue({"node", "user_agent_build_version", "version", "major_number"}, &major_number) &&
```
[the remark this answers is not in the document]
Comment (bochengchu) [reply extensions/common/context.cc]: yes, I was wrong, we should use istio version.

File: extensions/common/node_info.fbs [hunk 3]
```diff
@@ -45,6 +45,8 @@ table FlatNode {
   cluster_id:string;
   // instance ip addresses
   instance_ips:[string];
+  // proxy version
+  proxy_version:string;
```
Comment (kyessenov) [inline extensions/common/node_info.fbs:49]: Delete this?

File: extensions/common/istio_dimensions.h [hunk 4]
```diff
@@ -49,7 +49,8 @@ namespace Common {
   FIELD_FUNC(response_code)                                                                        \
   FIELD_FUNC(grpc_response_status)                                                                 \
   FIELD_FUNC(response_flags)                                                                       \
-  FIELD_FUNC(connection_security_policy)
+  FIELD_FUNC(connection_security_policy)                                                           \
+  FIELD_FUNC(proxy_version)
```
Comment (kyessenov) [inline extensions/common/istio_dimensions.h:53]: Delete this, too?

**Review by kyessenov — COMMENTED** [review]
Mostly LGTM except some superfluous code.

File: extensions/common/node_info.fbs (hunk 3 above)
Comment (bochengchu) [reply extensions/common/node_info.fbs]: Done.

File: extensions/common/istio_dimensions.h (hunk 4 above)
Comment (bochengchu) [reply extensions/common/istio_dimensions.h]: Done.

**Review by kyessenov — APPROVED** [review]

**Comment by ericvn** [comment]
@kyessenov Looking at the failure to merge Proxy updates into istio/istio: https://github.com/istio/istio/pull/45467 I see a figure in `TestStackdriverMonitoring`. I didn't see anything in the envoy proxy repo that would cause this, but I think this PR is suspect.

**Pull request merged**